crates/uv-interpreter/src/lib.rs

```diff
@@ -7,23 +7,20 @@
 //!   [`find_default_python`] when `--python` is used, [`find_default_python`] when `--system` is used
 //!   and the current venv by default.
 
-use std::ffi::OsString;
-use std::io;
-use std::path::PathBuf;
-use std::process::ExitStatus;
-
 use thiserror::Error;
 
+pub use crate::discovery::{
+    find_best_interpreter, find_default_interpreter, find_interpreter, Error as DiscoveryError,
+    InterpreterNotFound, InterpreterRequest, SourceSelector, SystemPython, VersionRequest,
+};
 pub use crate::environment::PythonEnvironment;
-pub use crate::find_python::{find_best_python, find_default_python, find_requested_python};
 pub use crate::interpreter::Interpreter;
-use crate::interpreter::InterpreterInfoError;
 pub use crate::python_version::PythonVersion;
 pub use crate::target::Target;
-pub use crate::virtualenv::{PyVenvConfiguration, VirtualEnvironment};
+pub use crate::virtualenv::{Error as VirtualEnvError, PyVenvConfiguration, VirtualEnvironment};
 
+mod discovery;
 mod environment;
-mod find_python;
 mod implementation;
 mod interpreter;
 pub mod managed;
```

Comment:
This feels like it could be private (with fetch-python moving a bit?)

Reply:
Do you mind if I tackle this separately? I'm trying not to touch the toolchain management stuff here and expect to revisit it when we add real support for it.

Reply:
Yeah that's a follow-up PR